src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -258,6 +258,14 @@ private bool ContainsPlaceholderScope(BoundValuePlaceholderBase placeholder)
         public override BoundNode? Visit(BoundNode? node)
         {
 #if DEBUG
+            VisitCore(node);
+#endif
+            return base.Visit(node);
+        }
+
+#if DEBUG
+        private void VisitCore(BoundNode? node)
```

Comment:
Given this method is only used for assertion purposes did you consider giving it an assert style named and making it conditional on DEBUG? For example: 

```csharp
[Conditional("DEBUG")]
private void AssertVisitCore(BoundNode? node)
```

Reply:
Making it just `Conditional` would fail to compile as it uses `_visited` field which is only available `#if DEBUG`.

Reply:
Can use an `#if DEBUG` within the method too. Personally I find the code easier to process when minimize the number of directives, particularly for call sites. Will defer to others here.

Reply:
My first choice in a situation like this will be to rely on conditional compilation vs. the attribute. It completely eliminates the method (including in produced metadata) for release build. Whereas the attribute doesn't have this effect. In my opinion the attribute is useful when we deal with a public API, which is not the case here.